Automattic/jetpack
# Introduce Mailchimp Shortcode

This PR introduces 2 things:
1. Mailchimp shortcode
2. Mailchimp gutenberg block.

More details in the master thread: p5uIfZ-8BV-p2

**The code behind this is already merged for WPCOM simple. If you want to test the UX - any simple site will work for proxied a12s**

## How this works

This enables visitors to sign up for a MailChimp list.

User needs to establish a Mailchimp connection in sharing section in calypso. There, they can use oauth flow to establish the connection and then select a list where subscribers will be saved.
( We imagine that this feature may be gated behing a plan - more info in the master thread )



### Shortcode

Once connection is established, shortcode `[jetpack-email-subscribe]` will display this signup form.

![screen shot 2018-11-12 at 18 14 01 pm](https://user-images.githubusercontent.com/3775068/48363970-76b06a00-e6a7-11e8-867b-e2be6927b1bf.png)

Once visitor signs up, her email is saved to a previously selected emaiil list in Mailchimp. We don't use it ourselves.
All email comes from the site owner through mailchimp

### Block

Once connection is established, site owner can use `jetpack/gutenberg` block.
If connection is NOT established, they get a prompt with the link to go to /sharing.


## Screenshots

### Block

![](https://user-images.githubusercontent.com/3775068/48910903-78c9b400-ee71-11e8-8a9d-ac85d5b11c87.png)

### Block, connection not set up


![zrzut ekranu 2018-12-28 o 15 30 05](https://user-images.githubusercontent.com/3775068/50518375-9e239080-0ab5-11e9-8639-ff887776c332.png)


## Testing instructions

1. Check out this PR on your Jetpack
2. Build blocks from calypso, the UI code is already merged
1. You need to log in as an Automattician
2. In /sharing you need to set up mailchimp connection and choose a list:
![screen shot 2018-11-12 at 18 13 20 pm](https://user-images.githubusercontent.com/3775068/48364106-d9096a80-e6a7-11e8-8513-5ea9279fd7e9.png)
3. Drop a `[jetpack-email-subscribe]` shortcode anywhere to test shortcode
4. Use `mailchimp` block for testing block.
5. Once you subscribe, your email will end up in the list set up in point 2.
